diff --git a/dist/emnapi-core.esm-bundler.js b/dist/emnapi-core.esm-bundler.js
index c6cfce7f55148e92cad21b024789f3b51a60fe63..3b7e9d0faed031ee65f42a928f95b8719afede29 100644
--- a/dist/emnapi-core.esm-bundler.js
+++ b/dist/emnapi-core.esm-bundler.js
@@ -4343,16 +4343,13 @@ function createNapiModule(options) {
                         viewDescriptor = { Ctor: Float16Array, address: external_data, length: byte_length >> 1, ownership: 1 /* ReferenceOwnership.kUserland */, runtimeAllocated: 0 };
                         break;
                     case -2 /* emnapi_memory_view_type.emnapi_buffer */: {
-                        if (!emnapiCtx.feature.Buffer) {
-                            throw emnapiCtx.createNotSupportBufferError('emnapi_create_memory_view', '');
-                        }
-                        viewDescriptor = { Ctor: emnapiCtx.feature.Buffer, address: external_data, length: byte_length, ownership: 1 /* ReferenceOwnership.kUserland */, runtimeAllocated: 0 };
+                        viewDescriptor = { Ctor: emnapiCtx.feature.Buffer || Uint8Array, address: external_data, length: byte_length, ownership: 1 /* ReferenceOwnership.kUserland */, runtimeAllocated: 0 };
                         break;
                     }
                     default: return envObject.setLastError(1 /* napi_status.napi_invalid_arg */);
                 }
                 var Ctor = viewDescriptor.Ctor;
-                var typedArray = typedarray_type === -2 /* emnapi_memory_view_type.emnapi_buffer */
+                var typedArray = typedarray_type === -2 /* emnapi_memory_view_type.emnapi_buffer */ && emnapiCtx.feature.Buffer
                     ? emnapiCtx.feature.Buffer.from(wasmMemory.buffer, viewDescriptor.address, viewDescriptor.length)
                     : new Ctor(wasmMemory.buffer, viewDescriptor.address, viewDescriptor.length);
                 var handle = emnapiCtx.addToCurrentScope(typedArray);
